contracts/consensus/Consensus.sol

```diff
@@ -593,31 +668,34 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
     )
         external
     {
-        address core = assignments[_metachainId];
+        CoreI core = assignments[_metachainId];
+
         require(
-            core != address(0),
+            core != CoreI(0),
             "Core does not exist for given metachain."
         );
 
         // Specified core must have creation lifetime status.
         require(
-            coreLifetimes[core] == CoreLifetime.creation,
+            coreLifetimes[address(core)] == CoreLifetime.creation,
```

Comment:
the mapping should probably be from `CoreI => CoreLifetime`?

Reply:
To my knowledge, solidity mapping key can be only elementary type.

Reply:
```
Mapping types use the syntax mapping(_KeyType => _ValueType) and variables of mapping type are declared using the syntax mapping(_KeyType => _ValueType) _VariableName. The _KeyType can be any built-in value type plus bytes and string. User-defined or complex types such as contract types, enums, mappings, structs or array types apart from bytes and string are not allowed. _ValueType can be any type, including mappings, arrays and structs.
```

https://solidity.readthedocs.io/en/v0.6.0/types.html#mapping-types